File: src/no/wtw/android/restserviceutils/RestServiceErrorObject.java
package no.wtw.android.restserviceutils;

import com.google.gson.annotations.SerializedName;

public class RestServiceErrorObject {

    @SerializedName("message")
    private String message = "Unknown error";
    @SerializedName("code")
    private int code = -1;
    @SerializedName("subCode")
    private int subCode = -1;

    public RestServiceErrorObject(String message) {
        this.message = message;
    }

    public String getMessage() {
        if (message == null)
            return "";
        return message;
    }

    @Override
    public String toString() {
        return code + "/" + code + " " + message != null ? message : "";
    }
}

File: library-core/src/main/java/no/wtw/android/restserviceutils/RestServiceErrorObject.kt
package no.wtw.android.restserviceutils

import com.google.gson.annotations.SerializedName

data class RestServiceErrorObject(
    @SerializedName("code")
    var code: Int?,
    @SerializedName("subCode")
    val subCode: Int = -1,
    @SerializedName("message")
    var message: String?,
) {

    override fun toString() = "$code/$code $message"

}